Fix auto complete enum string with quote on number

msivasubramaniaan opened the pull request:
### What does this PR do?

This PR adding double quotes to the number when enum type is string and it has number variable on it. 
Ex: "enum": ["12.1", "13.0", "13.1", "14.0", "all"]

### What issues does this PR fix or reference?
https://github.com/redhat-developer/yaml-language-server/issues/709

### Is it tested? How?
Yes, added UT

gorkem reviewed (changes requested):
This is not working as I expected. I used the following schema and all the completions were inserted with quotes. Looks like the insertText logic needs to be improved. 

```Json
{
  "type": "object",
  "additionalProperties": {
    "anyOf": [
      {
        "type": "array",
        "items": {
          "enum": ["12.1", 33, "34", 78],
          "type":  ["string", "integer"]
        }
      }
    ]
  }
} 
```

msivasubramaniaan commented:
> This is not working as I expected. I used the following schema and all the completions were inserted with quotes. Looks like the insertText logic needs to be improved.
> 
> ```json
> {
>   "type": "object",
>   "additionalProperties": {
>     "anyOf": [
>       {
>         "type": "array",
>         "items": {
>           "enum": ["12.1", 33, "34", 78],
>           "type":  ["string", "integer"]
>         }
>       }
>     ]
>   }
> } 
> ```

changed logic when the value type is integer and updated UT accordingly

gorkem commented on src/languageservice/services/yamlCompletion.ts:
```diff
@@ -1209,8 +1209,11 @@ export class YamlCompletion {
     }
-    type = Array.isArray(type) ? type[0] : type;
-    if (type === 'string') {
+    const types = Array.isArray(type) ? type : type;
```
What does this line try to achieve. It looks like it is assigning the same value to `types` regardless whether `type` is array or not.

msivasubramaniaan replied:
Added all the test cases for **number, string, Boolean, null and object**. 

> Also this method called when arrays having ENUM. So I have a question here: Is arrays of ENUM having another array in YAML? if so please provide me a reference schema for that as well. I will update my UT based on that.

gorkem reviewed (changes requested):
Should we check if this is handling all possible schema types correctly. Possible types are `string, number, integer, object, array, boolean, null`.
